internal/test/e2e/e2e_test.go

```diff
@@ -551,3 +571,159 @@ func persistenceTests(t *testing.T, kubectl *kubectlContext) {
 	require.NoError(t, err, out)
 	assert.Equal(t, expectedValue+"\n", out)
 }
+
+type probe struct {
+	name string
+}
+
+func (o *probe) Check(kubectl *kubectlContext) error {
+	out, err := kubectl.Logs(o.name, "probe")
+	lines := strings.Split(out, "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if len(line) == 0 {
+			continue
+		}
+		if !strings.HasPrefix(line, "SUCCESS") {
+			return fmt.Errorf("failure in probe logs: %s", out)
+		}
+	}
+	return err
+}
+
+const script = `
+set -o errexit
+set -o nounset
+set -o pipefail
+
+count=0
+while true
+do
+  if ! etcdctl --insecure-discovery --discovery-srv=%s --timeout=10s --total-timeout=11s set "/probe/${count}" "SUCCESS-${count}"
+  then
+	echo "FAIL-${count}"
+  fi
+  count=$(( count+1 ))
+  sleep 1
+done
+`
+
+func DeployEtcdProbe(kubectl *kubectlContext, name, discoveryName string) (*probe, error) {
+	obj := v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				{
+					Name:  "wait-for-cluster",
+					Image: "quay.io/coreos/etcd:v3.2.27",
```

Comment:
can we code the tag (or whole image, i don't mind) into a flag. seems like this would be forgotten and would end up drifting

Reply:
I've removed this probe test from the e2e tests. It was too flaky.